addons/mail/static/src/components/chat_window_header/chat_window_header.js

```diff
@@ -46,8 +50,21 @@ class ChatWindowHeader extends Component {
      * @param {MouseEvent} ev
      */
     _onClick(ev) {
-        const chatWindow = this.chatWindow;
-        this.trigger('o-clicked', { chatWindow });
+        if (!isEventHandled(ev, 'ChatWindowHeader.openProfile')) {
+            const chatWindow = this.chatWindow;
+            this.trigger('o-clicked', { chatWindow });
+        }
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickChatHeaderName(ev) {
+        if (this.chatWindow.thread && this.chatWindow.thread.correspondent) {
+            markEventHandled(ev, 'ChatWindowHeader.openProfile');
+            this.chatWindow.thread.correspondent.openProfile();
+        }
```

Comment:
For those 2 cases prefer inverting the condition and doing preventing `return;`

Then rest of the code can stay without extra indent and with minimal diff on it.